### PR TITLE
[#DevOps1] Refactored model and set up shared database

### PR DESCRIPTION
**Pull request by milicasiriski**
Learning resources:
- [Defining entity relationships](https://docs.microsoft.com/en-us/ef/core/modeling/relationships?tabs=fluent-api%2Cfluent-api-simple-key%2Csimple-key)
- [DbContextFactory](https://mehdi.me/ambient-dbcontext-in-ef6/)

### CODE REVIEW

File: HealthClinicBackendTests/HospitalRepositoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿using HealthClinicBackend.Backend.Repository.DatabaseSql;
+using Xunit;
+
+namespace HealthClinicBackendTests
+{
+    public class HospitalRepositoryTest
```
Comment (lukaDoric) [inline HealthClinicBackendTests/HospitalRepositoryTest.cs:6]: 💯

File: WebApplication/Backend/Repositorys/PatientRepository.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ private List<Patient> GetPatients(String sqlDml)
                 //(string)sqlReader[26];
                 entity.Guest = (bool)sqlReader[27];
                 entity.EmailConfirmed = (bool)sqlReader[28];
-                entity.ChosenDoctor = (string)sqlReader[29];
+                // entity.ChosenDoctor = (string)sqlReader[29];
```
Comment (lukaDoric) [inline WebApplication/Backend/Repositorys/PatientRepository.cs:68]: 👀

File: WebApplication/Backend/Repositorys/PhysitianRepository.cs [hunk 3]
```diff
@@ -18,23 +18,23 @@ public PhysitianRepository()
...
+                Physician entity = new Physician();
                 entity.SerialNumber = (string)sqlReader[0];
                 entity.Name = (string)sqlReader[1];
                 entity.Surname = (string)sqlReader[2];
-                entity.FullName = entity.Name + " " + entity.Surname;
-                entity.Id = (string)sqlReader[4];
-                entity.DateOfBirth = (DateTime)sqlReader[5];
-                entity.Contact = (string)sqlReader[6];
-                entity.Email = (string)sqlReader[7];
+                // entity.FullName = entity.Name + " " + entity.Surname;
```
Comment (lukaDoric) [inline WebApplication/Backend/Repositorys/PhysitianRepository.cs:33]: 👀

File: project-generated-code-backend/Backend/Model/Schedule/ProcedureType.cs [hunk 4]
```diff
@@ -9,111 +9,71 @@
...
+using System.Linq;
 
 namespace Model.Schedule
 {
     public class ProcedureType : Entity
     {
+        // private String name;
+        // private Specialization specialization;
+        // private int estimatedTimeInMinutes;
+        // private List<Equipment> requiredEquipment;
```
Comment (lukaDoric) [inline project-generated-code-backend/Backend/Model/Schedule/ProcedureType.cs]: 👀

File: project-generated-code-backend/Backend/Model/Util/City.cs [hunk 5]
```diff
@@ -6,95 +6,55 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
 
 namespace Model.Util
 {
     public class City : Entity
     {
-        private static int serialNumberGenerator = 0;
-        private String name;
-        private String postalCode;
+        // public List<Address> Address { get; set; }
```
Comment (lukaDoric) [inline project-generated-code-backend/Backend/Model/Util/City.cs]: 👀

**Review by lukaDoric — APPROVED** [review]

**Review by VladimirBudjen — APPROVED** [review]

**Comment by milicasiriski** [comment]
[(3) #DevOps1 Create generic postgresql repository](https://trello.com/c/56SiS5zn/38-3-devops1-create-generic-postgresql-repository)

**Pull request merged**